clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -712,6 +729,10 @@ private void close(Duration timeout, boolean swallowException) {
         if (applicationEventHandler != null)
             closeQuietly(() -> applicationEventHandler.close(timeout), "Failed to close application event handler with a timeout(ms)=" + timeout, firstException);
 
+        // Invoke all callbacks after the background thread exists in case if there are unsent async
+        // commits
+        maybeInvokeCallbacks();
```

Comment:
Why do we not check for fenced exceptions here?

The original consumer attempts to complete all in-flight async commit requests within `timeout` ms. Will closing the background thread take care of this?

Reply:
When the consumer is closing, we don't really need to throw a fenced id exception, because when the consumer is fenced we want the consumer to close right?

To your second question: yes, the work is inflight, KAFKA-15327.

Reply:
The fenced exception would indicate here that the last commit did not complete, correct? I think you are right, throwing the exception is not strictly required for correctness here; at the same time, when investigating what was going on the client during a fencing situation, it would be valuable information that the commit failed due to fencing, even if we are shutting down. If we can tell that from non-DEBUG logs, I'd say it's fine though, I haven't checked.